tests/unit/nupic/algorithms/nab_detector_test.py

```diff
@@ -0,0 +1,275 @@
+                        "permanenceInc": 0.1,
+                        "temporalImp": "cpp",
+                        "seed": 1960,
+                        "verbosity": 0
+                      },
+                      "anomalyParams": {
+                        "anomalyCacheRecords": None,
+                        "autoDetectThreshold": None,
+                        "autoDetectWaitRecords": 5030
+                      },
+                      "spEnable": True,
+                      "inferenceType": "TemporalAnomaly",
+                      "tpEnable": True
+                      },
+                    "predictAheadTime": None
+                  }
+    sensorParams = (modelParams["modelParams"]["sensorParams"]
+                               ["encoders"]["value"])
+    sensorParams["resolution"] = max(0.001,
+      (1.2 - 0.2) / sensorParams.pop("numBuckets"))
+    model = ModelFactory.create(modelParams)
+    
+    self.assertIs(type(model).__name__, "CLAModel", msg="The created model is "
+                  "not a CLAModel, but rather is of type %s" % type(model))
+
+
+  def testNABAnomalyLikelihood(self):
+    """
+    Tests the specific calls to nupic/algorithms/anomaly_likelihood as they"re
+    made in "NAB/detectors/numenta/numenta_detector.py".
+    Note "NAB/.../numenta_detector.py" has its own class AnomalyLikelihood,
+    different from nupic/algorithms/anomaly_likelihood.AnomalyLikelihood, but
+    which calls the functions estimateAnomalyLikelihoods() and 
+    updateAnomalyLikelihoods() from "nupic/algorithms/anomaly_likelihood.py".
+    """
+    # AnomalyLikelihood object initial values
+    iteration = 0
+    probationaryPeriod = 4
+    historicalScores = []
+    
+    likelihoodList = []
+    for dataPoint in self.data:
+      # Ignore the first probationaryPeriod data points
+      if len(historicalScores) < probationaryPeriod:
+        likelihood = 0.5
+      else:
+        if iteration % 4 == 0:
+          _, _, distribution = an.estimateAnomalyLikelihoods(
+                                 historicalScores,
+                                 skipRecords = probationaryPeriod)
+          likelihoods, _, distribution = an.updateAnomalyLikelihoods(
+                                [dataPoint], distribution)
+          likelihood = 1.0 - likelihoods[0]
+      historicalScores.append(dataPoint)
+      iteration += 1
+      likelihoodList.append(likelihood)
+    
+    truthLikelihoodList = [0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,
+                           0.044565462999999972, 0.044565462999999972,
+                           0.044565462999999972, 0.044565462999999972,
+                           0.90319951499999995, 0.90319951499999995,
+                           0.90319951499999995, 0.90319951499999995,
+                           0.78814460099999994, 0.78814460099999994,
+                           0.78814460099999994, 0.78814460099999994]
+    [self.assertAlmostEqual(likelihoodList[i], truthLikelihoodList[i],
```

Comment:
I suppose it is a matter of preference but I would do a regular for-loop since you aren't using the result of the comprehension.